MsixCore/msixmgr/msixmgr.cpp

```diff
@@ -192,26 +195,79 @@ int main(int argc, char * argv[])
 
     const HRESULT hrCreateRequest = cli.Init();
+
+    // Initializing all the variables for Telemetry
+    std::wstring WorkflowId, SourceApplicationId, CorrelationId, ErrorCode = L"", ErrorDesc = L"";
+    double WorkflowElapsedTime;
+    LARGE_INTEGER MsixMgrLoad_StartCounter, MsixMgrLoad_EndCounter, MsixMgrLoad_Frequency;
+    QueryPerformanceFrequency(&MsixMgrLoad_Frequency);
+
+    // Creating WorkflowId for tracking the Workflows
+    WorkflowId = msixmgr::CreateWorkflowId();
+    SourceApplicationId = cli.GetSourceApplicationId();    // SourceApplicationId indicates which application is executing the MSIXMGR commands. Useful during integration of MSIXMGR with other Tools.
+    CorrelationId = cli.GetCorrelationId();    // CorrelationId can be used to establish a correlation between MSIXMGR's Workflow and the parent application's Workflow, when MSIXMGR is being used in integration with any other Tool.
+
+    // Telemetry : Session Log
+    msixmgr::TraceLogSession(WorkflowId.c_str(), SourceApplicationId.c_str(), CorrelationId.c_str());
+
+    QueryPerformanceCounter(&MsixMgrLoad_StartCounter);
+
     if (SUCCEEDED(hrCreateRequest))
     {
         switch (cli.GetOperationType())
         {
         case OperationType::Add:
         {
+            // Telemetry : Add Workflow Log
+            msixmgr::TraceLogAddWorkflow(WorkflowId.c_str(), msixmgr::ExtractPackageNameFromFilePath(cli.GetPackageFilePathToInstall()).c_str());
+
+            HRESULT hr;
+
             AutoPtr<IPackageManager> packageManager;
-            RETURN_IF_FAILED(MsixCoreLib_CreatePackageManager(&packageManager));
+            hr = MsixCoreLib_CreatePackageManager(&packageManager);
+
+            if (FAILED(hr))
+            {
+                ErrorCode = msixmgr::GetErrorCodeFromHRESULT(hr);
+                ErrorDesc = L"Failed creation of Package Manager Object. HRESULT " + ErrorCode + L".";
+
+
+                // Telemetry : Workflow Log
+                QueryPerformanceCounter(&MsixMgrLoad_EndCounter);
+                WorkflowElapsedTime = msixmgr::CalcWorkflowElapsedTime(MsixMgrLoad_StartCounter, MsixMgrLoad_EndCounter, MsixMgrLoad_Frequency);
+                msixmgr::TraceLogWorkflow(WorkflowId.c_str(), L"Add", false, WorkflowElapsedTime, ErrorCode.c_str(), ErrorDesc.c_str());
+
+                RETURN_IF_FAILED(hr);
+            }
 
             if (cli.IsQuietMode())
             {
                 if (!isAdmin)
                 {
+                    ErrorCode = L"";
+                    ErrorDesc = L"Relaunching as Admin.";
+
+                    // Telemetry : Workflow Log
+                    QueryPerformanceCounter(&MsixMgrLoad_EndCounter);
+                    WorkflowElapsedTime = msixmgr::CalcWorkflowElapsedTime(MsixMgrLoad_StartCounter, MsixMgrLoad_EndCounter, MsixMgrLoad_Frequency);
+                    msixmgr::TraceLogWorkflow(WorkflowId.c_str(), L"Add", false, WorkflowElapsedTime, ErrorCode.c_str(), ErrorDesc.c_str());
+

```

Comment:
I think for a user, the flow can run into this block unknowingly many a times, because users generally forget to run the terminal in Admin mode. Logging in this case is not giving us any relevant metrics. I think we can just stop the timer here and not log anything.

Reply:
As per conversation with PM, this could be a useful information and no harm in capturing this information